pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/HashJoinOperator.java

```diff
@@ -101,27 +114,32 @@ private void buildBroadcastHashTable() {
         }
         rightBlock = _rightTableOperator.nextBlock();
       }
+      if (rightBlock.isErrorBlock()) {
+        _upstreamErrorBlock = (MetadataBlock) rightBlock.getDataBlock();
+      }
       _isHashTableBuilt = true;
     }
   }
 
   private TransferableBlock buildJoinedDataBlock(TransferableBlock leftBlock)
       throws Exception {
-    if (TransferableBlockUtils.isEndOfStream(leftBlock)) {
-      return TransferableBlockUtils.getEndOfStreamTransferableBlock();
-    }
-    List<Object[]> rows = new ArrayList<>();
-    _leftTableSchema = leftBlock.getDataSchema();
-    _resultRowSize = _leftTableSchema.size() + _rightTableSchema.size();
-    List<Object[]> container = leftBlock.getContainer();
-    for (Object[] leftRow : container) {
-      List<Object[]> hashCollection =
-          _broadcastHashTable.getOrDefault(_leftKeySelector.computeHash(leftRow), Collections.emptyList());
-      for (Object[] rightRow : hashCollection) {
-        rows.add(joinRow(leftRow, rightRow));
+    if (!TransferableBlockUtils.isEndOfStream(leftBlock)) {
+      List<Object[]> rows = new ArrayList<>();
+      List<Object[]> container = leftBlock.getContainer();
+      for (Object[] leftRow : container) {
+        List<Object[]> hashCollection = _broadcastHashTable.getOrDefault(
+            _leftKeySelector.computeHash(leftRow), Collections.emptyList());
+        for (Object[] rightRow : hashCollection) {
+          rows.add(joinRow(leftRow, rightRow));
+        }
       }
+      return new TransferableBlock(rows, computeSchema(), BaseDataBlock.Type.ROW);
+    } else if (leftBlock.isErrorBlock()) {
+      _upstreamErrorBlock = (MetadataBlock) leftBlock.getDataBlock();
+      return TransferableBlockUtils.repackErrorBlock(_upstreamErrorBlock);
+    } else {
+      return new TransferableBlock(DataBlockUtils.getEmptyDataBlock(_resultSchema));
```

Comment:
If we reach here should we assert that leftBlock is a `EOS` type `METADATA` block ?

Reply:
echo back on https://github.com/apache/pinot/pull/9064#discussion_r928010027. yes this will be fixed.